Package.swift

```diff
@@ -7,23 +7,39 @@ let package = Package(
     products: [
         .executable(
             name: "nio-apns-example",
-            targets: ["NIOAPNSExample"]),
+            targets: ["NIOAPNSExample"]
+        ),
         .library(
             name: "NIOAPNS",
-            targets: ["NIOAPNS"]),
+            targets: ["NIOAPNS"]
+        ),
     ],
     dependencies: [
-      .package(url: "https://github.com/moritzsternemann/nio-h2", .upToNextMinor(from: "0.1.0")),
+        .package(
+            url: "https://github.com/moritzsternemann/nio-h2",
+            .upToNextMinor(from: "0.1.0")
+        ),
+        .package(
+            url: "https://github.com/IBM-Swift/OpenSSL",
```

Comment:
I think it would be better to have this as an internal dependency. Something like `CAPNSOpenSSL`. You can copy what vapor/crypto does: https://github.com/vapor/crypto/tree/master/Sources/CCryptoOpenSSL

Reply:
Oh yeah, that's better.. will change

Reply:
@moritzsternemann My original PR was relying on just open SSL headers
```
.systemLibrary(
	            name: "OpenSSL",
	            pkgConfig: "openssl",
	            providers: [
	                .apt(["openssl libssl-dev"]),
	                .brew(["openssl"]),
	            ]
	        ),
```

Reply:
yep, that's the 'new' Swift 4.2 way to import system libraries. Anything wrong with this other than maybe backwards compatibility @tanner0101?
[SE-0208](https://github.com/apple/swift-evolution/blob/master/proposals/0208-package-manager-system-library-targets.md) describes this a lot better than [the actual documentation](https://github.com/apple/swift-evolution/blob/master/proposals/0208-package-manager-system-library-targets.md)

Reply:
@moritzsternemann if that works, then seems perfectly fine to me 👍. Having the separate module can be useful if you need to write any C code, but seems like you don't for this use case. vapor/crypto on the other hand needs it because we support both OpenSSL 1.0 and 1.1, and the least painful way to do that is with [C shims](https://github.com/vapor/crypto/blob/master/Sources/CCryptoOpenSSL/shim.c).

Reply:
(Btw, sent you a message on Discord :))